.github/workflows/release.yml

```diff
@@ -3,36 +3,53 @@ name: release
 on:
   push:
     tags:
-      - v0.*
-      - v1.*
+      - "v*"
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: tag (ex. v1.0) to publish
+        required: false
 
 jobs:
-  release:
-    runs-on: snowbridge-runner
-    env:
-      CARGO_INCREMENTAL: 0
-      RUST_BACKTRACE: 1
+  release-binary:
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v1
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v4
         with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('parachain/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: nightly-2023-04-22
-          target: wasm32-unknown-unknown
+          go-version: '1.20'
+      - name: Install tools
+        run: >
+          go install github.com/magefile/mage@v1.15.0 && 
+          go install github.com/ferranbt/fastssz/sszgen@v0.1.3
       - name: build
-        uses: actions-rs/cargo@v1
-        with:
-          command: build
-          args: --manifest-path parachain/Cargo.toml --workspace --release
+        working-directory: relayer
+        run: mage build
       - uses: actions/upload-artifact@v1
         with:
-          name: snowbridge-node
-          path: parachain/target/release/snowbridge
+          name: snowbridge-relay
+          path: relayer/build/snowbridge-relay
+
+  release-docker-image:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2.2.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push
+        run: |
+          docker buildx build \
+          --platform linux/amd64 \
+          -f relayer/Dockerfile \
+          -t "ronyang/snowbridge-relay:${{ github.event.inputs.tag || github.sha }}" \
```

Comment:
@vgeddes Except for the binary format I would suggest we add docker image as another option for running relay service. This is also the recomended way for running most of the other components in our stack.

Problem is that I don't have authority to config the repo so use my own account here temporarily. We need to register an account in docker hub as orgnization and add secrets here so to push image.

Reply:
Actually in the past we've used GitHub Packages as a docker repository: https://github.com/Snowfork/snowbridge/pkgs/container/snowbridge-relay

Can you try and upload the image there to test?

Reply:
Cool! Seems also require to config `GITHUB_TOKEN` for [Authenticating to the Container registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry), so is that already configured in this repo before?

Reply:
Some revamp in https://github.com/Snowfork/snowbridge/pull/881/commits/81f3bf07519bdc432ed55168128edd747d131415 also add a [workflow_dispatch](https://github.com/Snowfork/snowbridge/blob/765709754952dcabd263ca3ece19af5fd3987a6c/.github/workflows/release.yml#L7-L11) for manually trigger the action for test.

But seems without first merging to main it's not easy to test the newly added action.